core/shared/src/main/scala/com/lynbrookrobotics/potassium/Component.scala

```diff
@@ -31,8 +38,17 @@ abstract class Component[T] {
     if (controller.expectedPeriodicity == NonPeriodic) {
       throw new IllegalArgumentException("Controller must be periodic")
     }
+    currentTimingHandle = Some(
+      controller.zipWithDt.foreach{ case (_, dt: Time) =>
+        histogram.accept(dt.toMilliseconds)
+        if (controller.expectedPeriodicity.asInstanceOf[Periodic].period > 2 * dt ) {
+          println(histogram.toString)
+        }
+     }
+    )
 
     currentControllerHandle.foreach(_.apply())
+    currentTimingHandle.foreach(_.apply())
```

Comment:
I think you meant to kill the previous handle and reset it.
You actually set `currentTimingHandle` and then killed it right away.